plutus-ledger/src/Ledger/Constraints/TxConstraints.hs

```diff
@@ -202,15 +202,26 @@ mustPayToOtherScript vh dv vl =
 {-# INLINABLE mustMintValue #-}
 -- | Create the given value
 mustMintValue :: forall i o. Value -> TxConstraints i o
-mustMintValue = foldMap valueConstraint . (AssocMap.toList . Value.getValue) where
+mustMintValue = mustMintValueWithRedeemer (Redeemer $ PlutusTx.toData ())
+
+{-# INLINABLE mustMintValueWithRedeemer #-}
+-- | Create the given value
+mustMintValueWithRedeemer :: forall i o. Redeemer -> Value -> TxConstraints i o
+mustMintValueWithRedeemer red = foldMap valueConstraint . (AssocMap.toList . Value.getValue) where
     valueConstraint (currencySymbol, mp) =
         let hs = Value.currencyMPSHash currencySymbol in
-        foldMap (Haskell.uncurry (mustMintCurrency hs)) (AssocMap.toList mp)
+        foldMap (Haskell.uncurry (mustMintCurrencyWithRedeemer hs red)) (AssocMap.toList mp)
 
 {-# INLINABLE mustMintCurrency #-}
 -- | Create the given amount of the currency
 mustMintCurrency :: forall i o. MintingPolicyHash -> TokenName -> Integer -> TxConstraints i o
-mustMintCurrency mps tn = singleton . MustMintValue mps tn
+-- TODO: pass up?
```

Comment:
@michaelpj What did you mean with this TODO?

Reply:
I cheated and just passed `()` as the redeemer here: arguably we should expose this choice to the user and have another argument here. But that probably leads to lots of piping this up to the top level, which we probably should do but maybe not now.

Reply:
I added `mustMintCurrencyWithRedeemer`, so I think that covers this TODO?

Reply:
But we should also audit the callers of `mustMintCurrency`. If it's called by a user, that's fine (since we have no usecases that need reedeemers yet), but if it's called by another function then we might need a variant of _that_ function that handles redeemers.

Reply:
Yes, I checked that. I also added `mustMintValueWithRedeemer`. `mustMintValue` and `mustMintCurrency` are only used in `plutus-use-cases`.

Reply:
Way ahead of me :)